### PR TITLE
feat: Add onboarding guidance DOCS-470

### PR DESCRIPTION
**Pull request by prcr**
Expands the "Codacy quickstart" into a guided path to cover the steps for onboarding a new repository by reusing and refactoring the sections already available on the page [How do I block merging pull requests using Codacy as a quality gate?](https://docs.codacy.com/faq/general/how-do-i-block-merging-prs-using-codacy-as-a-quality-gate/).

Supersedes https://github.com/codacy/docs/pull/1547.

### :eyes: Live preview
- https://feat-expand-codacy-quickstart-take2--docs-codacy.netlify.app/getting-started/codacy-quickstart/
- https://feat-expand-codacy-quickstart-take2--docs-codacy.netlify.app/getting-started/configuring-your-repository/
- https://feat-expand-codacy-quickstart-take2--docs-codacy.netlify.app/getting-started/integrating-codacy-with-your-git-workflow/

### :construction: To do
- [x] @nicklem: Edit the intros for "Configuring your repository" and "Integrating Codacy with your Git workflow"
- [x] Improve meta description for new pages
- [x] Fix TODOs
- [x] Consider pinging Success (Mariana Pacheco?) for a review in the context of [some recent related work with a customer (Slack)](https://codacy.slack.com/archives/CA5GWEXSN/p1674218247760999)

### CODE REVIEW

File: docs/getting-started/integrating-codacy-with-your-git-workflow.md
Comment (nicklem) [inline docs/getting-started/integrating-codacy-with-your-git-workflow.md:6]: I'd consider dropping this sentence. We mention it further along in a tip admonition under point 1

Comment (prcr) [reply docs/getting-started/integrating-codacy-with-your-git-workflow.md:6]: I agree that this is too much detail for the intro. :+1:

However, there's some extra info here that would complement the tip below really well:

- The tip makes a big jump between "the defaults help maintain the current code quality of your repository" and "focus on stopping new critical issues from entering your code base".
- This sentence provides the extra context necessary to understand why it's unadvisable for customers that don't have good enough code quality in their repositories or mature team practices yet to try to follow such strict quality gates. It would basically help glue together the two paragraphs in the tip. :nerd_face:

File: docs/getting-started/configuring-your-repository.md
Comment (prcr) [inline docs/getting-started/configuring-your-repository.md:9]: What if we mentioned the false positives only at the end of this paragraph?

That way we could have a more straightforward sentence flow because we would mention everything related to matching the use cases of the team (coding conventions, best practices, and quality standards to enforce), and only then mention the importance of avoiding false positives. I feel that like this, the paragraph is switching back and forth between the two things.

**Review by prcr — COMMENTED** [review]
Good edits, I feel like now we're going in a more focused direction! :smile:

File: docs/getting-started/integrating-codacy-with-your-git-workflow.md
Comment (nicklem) [reply docs/getting-started/integrating-codacy-with-your-git-workflow.md]: Agreed. I integrated the sentence midway and polished for flow

File: docs/getting-started/configuring-your-repository.md
Comment (nicklem) [reply docs/getting-started/configuring-your-repository.md]: OK, I tried reversing the sentence order

File: docs/getting-started/configuring-your-repository.md
Comment (prcr) [inline docs/getting-started/configuring-your-repository.md:3]: Here are some suggestions to round things off:

- The "use cases of the team" (unless we have some other ideas), are actually the existing conventions or best practices of the team, and any new ones that the user configuring the repository may want to introduce. So we can rephrase to include these as examples instead.
- I like your idea of "promoting" best practices/conventions, and I think we can replace "enforce" with that since it has a less negative connotation.
- Finally, I suggest that we make it more explicit why the false positives can be distracting and time-consuming. This is the kind of argumentation that should convince the users not to skip performing these configurations.

```suggestion
Once you've [added your first repository](codacy-quickstart.md), it's important that you configure the analysis tools to match the use cases of your team, such as configuring any coding conventions and best practices that your team may already be following or that you want to promote. It's also critical to review the configurations to avoid reporting false positives or any other issues that don't bring value to your team, which can introduce unwanted delays to the development process.
```

**Review by prcr — COMMENTED** [review]
I suggest some more clarifications to one of the intros, but this stage of the refactoring is looking good! :ok_hand: 

Next in line for review before we consider merging these changes are the links that previously pointed to the page "How do I block ..." but right now are pointing to the page "Integrating Codacy with your Git workflow". You can start having a look and tomorrow we'll figure out what to do with them. :rocket:

File: docs/assets/includes/status-checks-important.md
Comment (nicklem) [inline docs/assets/includes/status-checks-important.md:8]: I worded this as a subsequent step after the user does the tasks above, although there are other ways besides coverage to block pull requests. I believe that this is OK, given the clear context.

File: docs/repositories-configure/adjusting-quality-settings.md
Comment (nicklem) [inline docs/repositories-configure/adjusting-quality-settings.md:18]: I left out the mention of toggling Diff coverage is under or Coverage variation is under, since we now mention it explicitly on the target page.

File: docs/getting-started/integrating-codacy-with-your-git-workflow.md
Comment (nicklem) [inline docs/getting-started/integrating-codacy-with-your-git-workflow.md:42]: What are out main points here? Something close to [the admonition above](https://github.com/codacy/docs/pull/1584/files#diff-999addb93ee2b841e94ab56f79160505a69c497ce464c762f2ea347a22d6a481R18)?

I'm thinking:
* Ensure your setup works (we mention this already)
* Test it out for a while before you turn the switch and let it block you

File: docs/repositories-configure/adjusting-quality-settings.md
Comment (prcr) [reply docs/repositories-configure/adjusting-quality-settings.md:18]: Yes, makes sense!

File: docs/assets/includes/status-checks-important.md
Comment (prcr) [reply docs/assets/includes/status-checks-important.md:8]: The way you reworded the sentence looks good :ok_hand: but now that you mention this, I find it quite weird that the admonition which had the main goal of instructing how to obtain status checks for coverage was also referencing the blocking of pull requests, since the two things are not necessarily dependent on one another. :thinking: 

While checking out the rendered page to see everything in context, I believe that the best place for this sentence is actually right after the explanation of what the status checks are, like this:

image

This has the advantage of giving a bit more visibility to the fact that status checks optionally allow blocking the merge of pull requests, and keeping this reference separated from the coverage-related admonition.

File: docs/getting-started/integrating-codacy-with-your-git-workflow.md
Comment (prcr) [reply docs/getting-started/integrating-codacy-with-your-git-workflow.md:46]: Indeed, the key information here is making sure that the user is "happy" with their configuration so that they don't inadvertently block the work of the development teams.

All of this is already conveyed in the `important` admonition below, so I would experiment with transforming the admonition into the intro. We can round it off a bit better with:

* An introductory sentence explaining what it means or why it can be a good idea to block merging pull requests
* Changing the "Finally, follow the instructions [...]" to something along the lines of "Once you've tested out Codacy for a while and you're happy with the level of feedback provided, follow the instructions [...]"

Comment (prcr) [reply docs/getting-started/integrating-codacy-with-your-git-workflow.md:46]: Besides these improvements, now I'm definitely convinced that we must mark the section for blocking the merge of pull requests as optional since not all customers will want to do it - and indeed, most customers **won't be ready** to do it during the onboarding.

**Review by prcr — COMMENTED** [review]
Cool changes! :nerd_face: I left two suggestions/ideas to help close the remaining open items. :rocket:

File: docs/assets/includes/status-checks-important.md
Comment (nicklem) [reply docs/assets/includes/status-checks-important.md:8]: Sopunds good. This also simplifies the import a bit, getting rid of the conditional.

Comment (prcr) [reply docs/assets/includes/status-checks-important.md]: I hadn't thought of that, but that's a nice simplification indeed! :star_struck:

**Comment by nicklem** [comment]
@prcr The last TODO is ready for review:

File: docs/getting-started/integrating-codacy-with-your-git-workflow.md
Comment (prcr) [inline docs/getting-started/integrating-codacy-with-your-git-workflow.md:51]: There's some repetition between "it's therefore important" and "this is important". One way to improve this could be to mention the false positives directly on the sentence introducing the list, for example:

> It's therefore important to eliminate any false positives that could inadvertently block the work of your team before activating this feature:

Comment (prcr) [reply docs/getting-started/integrating-codacy-with-your-git-workflow.md:51]: Now that I see this version, I even think we could bring back the `important` admonition to give more visibility just to this part:

```
!!! important
    It's important to eliminate any false positives that could inadvertently block the work of your team before activating this feature:
    
    - Validate [...]
    - Double-check [...]
```

What do you think?

**Review by prcr — COMMENTED** [review]
It's looking great @nicklem! I left some final suggestions, but I'll change the pull request to **Open** and we'll wait for @claudiacarpinteiro to have a look.

File: docs/getting-started/integrating-codacy-with-your-git-workflow.md
Comment (nicklem) [reply docs/getting-started/integrating-codacy-with-your-git-workflow.md]: Good idea. I moved the sentences around and reworded them for flow. Let me know 😄

File: docs/getting-started/integrating-codacy-with-your-git-workflow.md
Comment (nicklem) [inline docs/getting-started/integrating-codacy-with-your-git-workflow.md:46]: An alternative here:
```suggestion
Once you've tested out Codacy for a while and you're happy with the level of feedback provided, you can decide to enforce the quality gates and use Codacy to block merging pull requests on your Git provider. This is the best way to protect your code from unwelcome changes and fully integrates code quality and coverage analysis into your development pipeline.
```

Comment (prcr) [reply docs/getting-started/integrating-codacy-with-your-git-workflow.md:46]: Good idea, I guess we need to start being more explicit about quality [and|or|versus] coverage. :+1:

**Comment by prcr** [comment]
I can't approve my own pull request, but from my side this is looking good for a merge. :heavy_check_mark: 

Let's just wait for a sanity check from a couple of other reviewers.

File: docs/getting-started/codacy-quickstart.md
Comment (claudiacarpinteiro) [inline docs/getting-started/codacy-quickstart.md:56]: A small suggestion to allow the reader to read all the sentence before they get distracted by the link.

```suggestion
The first analysis is based on default tool and pattern configurations. It's now important that you configure your repository to integrate code analysis seamlessly into your existing pipeline. See how to [configure your repository to match the use cases of your team](configuring-your-repository.md).
```

File: docs/getting-started/configuring-your-repository.md
Comment (claudiacarpinteiro) [inline docs/getting-started/configuring-your-repository.md:2]: Are we configuring the "repository" or the "analysis tool"? I'm seeing both expressions used across this content. Do we need to clarify this? 🤔

File: docs/getting-started/integrating-codacy-with-your-git-workflow.md
Comment (claudiacarpinteiro) [inline docs/getting-started/integrating-codacy-with-your-git-workflow.md:17]: The configuration of the quality gate rules is only necessary for the integration with the Git workflow? Isn't it necessary besides this integration?

**Review by claudiacarpinteiro — COMMENTED** [review]
The flow looks very good! Nice work! 👍 
I only left a small suggestion and some questions on details that were not clear to me.

File: docs/getting-started/integrating-codacy-with-your-git-workflow.md
Comment (nicklem) [reply docs/getting-started/integrating-codacy-with-your-git-workflow.md:17]: To enforce quality gates, you must also integrate Codacy with the Git provider. Moreover, the Git provider only blocks PRs if the quality gates are set. So this step and the following one are very correlated.

You can also set quality gates with no integration, and this will display warnings on the Codacy UI, but it won't block the development workflow.

File: docs/getting-started/configuring-your-repository.md
Comment (nicklem) [reply docs/getting-started/configuring-your-repository.md]: Good point. I tweaked the wording

File: docs/organizations/using-a-coding-standard.md
Comment (prcr) [inline docs/organizations/using-a-coding-standard.md:96]: I think this change was pushed by mistake @nicklem:

```suggestion
```

Comment (nicklem) [reply docs/organizations/using-a-coding-standard.md:96]: You're right. I merged recently and this somehow snuck in. Reverting

**Review by nicklem — APPROVED** [review]
At long last! :shipit:

**Pull request merged**
